docs/rst/manual/reference/configuration/magic_strings.rst

```diff
@@ -0,0 +1,242 @@
+
+Magic strings
+=============
+
+ERT exposes some magic strings that can be used in your configuration file.
+These strings will at runtime be replaced with corresponding values, and
+can be useful if you need to use information such as file paths or experiment
+information such as realization or iteration number in your workflows or jobs.
+
+Example:
+::
+
+   YOUR_WORKFLOW_JOB <RUNPATH_FILE> <CONFIG_PATH>
+
+Will be replaced with something similar to the following at runtime
+
+::
+
+   YOUR_WORKFLOW_JOB /path/to/the/runpath/file/.ert_runpath_list /path/to/config/
+
+In the following we separate between :ref:`experiment constants <ms_exp_constants>`,
+:ref:`experiment variables <ms_exp_variables>` and :ref:`experiment dependent variables <ms_exp_dependent_variables>`.
+Experiment are magic strings that do not change based on the iteration or realization number,
+while experiment variables will typically be replaced with different values depending on
+the current iteration or realization index. The dependent variables will also change,
+but the changes directly depend on the changes in the experiment variables.
```

Comment:
I was confused at exactly what this distinction meant. The conversation below cleared it up though. If I understood correctly, perhaps make it clear that they might change, if the user uses a variable in their definition.

Reply:
I agree, updated the comment now!